### ISSUE TITLE
SOSL parsing error 2

### ISSUE THREAD

**Issue by xixiaofinland** [aheber/tree-sitter-sfapex#55]
```
class SOSLClass {
  void check() {
    searchResult = [
      FIND 'Hello'
      IN ALL FIELDS
      RETURNING Account(Id, Name USING ListView(id = listViewId))
    ];
  }
}
```

**Comment by xixiaofinland** [comment]
Same goes to this.

```
class SOSLClass {
    void simpleSOSL() {
    String a = 'Hello';
    List<List<SObject>> searchResult = [FIND 'FirstLastName'];
    searchResult = [FIND :a];
    searchResult = [FIND :a USING PHRASE SEARCH];
    searchResult = [FIND :a USING ADVANCED SEARCH];
  }
}
```

**Comment by aheber** [comment]
I don't think this is valid SOSL, built as a raw SOSL query instead of nested in Apex, shouldn't make a significant difference for the clause as far as I can tell, just easier to review in isolation.

```
FIND {Hello}
      IN ALL FIELDS
      RETURNING Account(Id, Name USING ListView(id = listViewId))
```

If I run it or things similar to it in an org then it fails to execute. Depending on which variation I run I get different errors.
```
FIND {Hello}
      IN ALL FIELDS
      RETURNING Account(Id, Name USING ListView(Id = Recent))
```

Produces
`Invalid using expr`

Using an Id (inside or outside of quotes)
```
FIND {Hello}
      IN ALL FIELDS
      RETURNING Account(Id, Name USING ListView(Id = '00BRL000008aC1W2AU'))
```

Produces
```
Account(Id, Name USING ListView(Id = '00BRL000008aC1W2AU'))
                                    ^
ERROR at Row:3:Column:53
Malformed search
```

This is probably on the fringe of "parsable but not executable" so I need to decide if I want to add support to parse it even though as far as I can tell it would never be valid in Apex or run in the org.

I'll have to think about it but I lean toward not going out of my way to add support, unless you can get me an example that does execute against an org.

**Comment by aheber** [comment]
> class SOSLClass {
>     void simpleSOSL() {
>     String a = 'Hello';
>     List<List<SObject>> searchResult = [FIND 'FirstLastName'];
>     searchResult = [FIND :a];
>     searchResult = [FIND :a USING PHRASE SEARCH];
>     searchResult = [FIND :a USING ADVANCED SEARCH];
>   }
> }

I'm going to move this to another issue. I think these are separate and as far as I can tell they are valid but not documented.
